model-optimizer/extensions/front/tf/CTCGreedyDecoderReplacement.py

```diff
@@ -0,0 +1,145 @@
+        replace_ctc_greedy_decoder(graph, match)
+
+
+class CTCGreedyDecoderWithSparseToDenseShapeReplacement(FrontReplacementSubgraph):
+    """
+    TensorFlow CTCGreedyDecoder produces output in a sparse tensor that is not supported by Inference Engine, and
+    Inference Engine's CTCGreedyDecoderSeqLen has a different output that is in a dense format. So this transformation
+    intents to replace TF CTCGreedyDecoder+SparseToDense where SparseToDense third input get from CTCGreedyDecoder
+    second output to CTCGreedyDecoderSeqLen which compatible with IE.
+    """
+    enabled = True
+
+    @staticmethod
+    def pattern(**kwargs):
+        return dict(
+            nodes=[('decoder', dict(op='CTCGreedyDecoderSeqLen', output_sparse_format=True)),
+                   ('cast', dict(op='Cast')),
+                   ('sparse_to_dense', dict(op='SparseToDense'))
+                   ],
+            edges=[('decoder', 'sparse_to_dense', {'out': 0}),
+                   ('decoder', 'cast', {'out': 1}),
+                   ('decoder', 'sparse_to_dense', {'out': 2}),
+                   ('cast', 'sparse_to_dense', {'out': 0})]
+        )
+
+    def replace_sub_graph(self, graph: Graph, match: dict):
+        replace_ctc_greedy_decoder(graph, match)
+
+
+class CTCGreedyDecoderSingleReplacement(FrontReplacementPattern):
+    """
+    TensorFlow CTCGreedyDecoder produces output in a sparse tensor that is not supported by Inference Engine, and
+    Inference Engine's CTCGreedyDecoderSeqLen has a different output that is in a dense format. So this transformation
+    handles a single TF CTCGreedyDecoder and warns the user about another format of the output
+    """
+    enabled = True
+
+    def run_after(self):
+        return [CTCGreedyDecoderReplacement, CTCGreedyDecoderWithSparseToDenseShapeReplacement]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        for ctc_greedy_decoder_tf in graph.get_op_nodes(op='CTCGreedyDecoderSeqLen', output_sparse_format=True):
+            ctc_greedy_decoder_tf_name = ctc_greedy_decoder_tf.soft_get('name', ctc_greedy_decoder_tf.id)
+
+            for port_num in range(4):  # TF CTCGreedyDecoder have 4 output tensors
+                if port_num in ctc_greedy_decoder_tf.out_ports():
+                    if ctc_greedy_decoder_tf.out_port(port_num).disconnected():
+                        # Create Result operation and connect it to the output if it is not connected
+                        # to any other operation
+                        second_result = Result(graph,
+                                               {'name': ctc_greedy_decoder_tf_name + '/seq_lengths_output'}
+                                               ).create_node()
+                        ctc_greedy_decoder_tf.out_port(1).connect(second_result.in_port(0))
+                    elif ctc_greedy_decoder_tf.out_port(port_num).get_destination().node.soft_get('op') != 'Result':
+                        return
```

Comment:
it is not good because you can corrupt graph with new Result nodes in advance if the transformation is not applicable. Please strictly separate these steps:
1. Check that all connected inputs are `Result` node, otherwise the transformation - not applicable
2. Add `Result` node to any 0, 1 output ports. And name the with different names not ctc_greedy_decoder_tf_name + '/seq_lengths_output'. Please use more meaningful names ctc_greedy_decoder_tf_name + '/decoded_classes' and ctc_greedy_decoder_tf_name + '/sequence_length'
3. Warning and transpose adding
4. Remove extra outputs from 2, 3 output ports

Reply:
Done, transformation conditions updated.